Analysis/Tasks/PWGHF/HFXicCandidateSelector.cxx

```diff
@@ -0,0 +1,447 @@
+      auto trackNeg1 = hfCandProng3.index1_as<aod::BigTracksPID>(); //negative daughter (positive for the antiparticles)
+      auto trackPos2 = hfCandProng3.index2_as<aod::BigTracksPID>(); //positive daughter (negative for the antiparticles)
+
+      topolXicpKpi = true;
+      topolXicpiKp = true;
+      pidXicPKPi = -1;
+      pidXicPiKP = -1;
+      kaonNeg = -1;
+      prot1 = -1;
+      prot2 = -1;
+      pion1 = -1;
+      pion2 = -1;
+      //proton = -1;
+      //kaonMinus = -1;
+      //pionPlus = -1;
+
+      // daughter track validity selection
+      if (!daughterSelection(trackPos1) || !daughterSelection(trackNeg1) || !daughterSelection(trackPos2)) {
+        hfSelXicCandidate(statusXicpKpi, statusXicpiKp);
+        continue;
+      }
+
+      //implement filter bit 4 cut - should be done before this task at the track selection level
+
+      //conjugate independent topological selection
+      if (!selectionTopol(hfCandProng3)) {
+        hfSelXicCandidate(statusXicpKpi, statusXicpiKp);
+        continue;
+      }
+
+      //conjugate dependent topplogical selection for Xic
+
+      topolXicpKpi = selectionTopolConjugate(hfCandProng3, trackPos1, trackNeg1, trackPos2);
+      topolXicpiKp = selectionTopolConjugate(hfCandProng3, trackPos2, trackNeg1, trackPos1);
+
+      if (!topolXicpKpi && !topolXicpiKp) {
+        hfSelXicCandidate(statusXicpKpi, statusXicpiKp);
+        continue;
+      }
+
+      /*
+      proton = selectionPID(trackPos1, 2212);
+      kaonMinus = selectionPID(trackNeg1, 321);
+      pionPlus = selectionPID(trackPos2, 211);
+
+      if (proton == 0 || kaonMinus == 0 || pionPlus == 0) {
+        pidLc = 0; //exclude Lc
+      }
+      if (proton == 1 && kaonMinus == 1 && pionPlus == 1) {
+        pidLc = 1; //accept Lc
+      }
+
+      if (pidLc == 0) {
+        hfSelLcCandidate(statusLcpKpi, statusLcpiKp);
+        continue;
+      }
+
+      if ((pidLc == -1 || pidLc == 1) && topolLcpKpi) {
+        statusLcpKpi = 1; //identified as Lc
+      }
+      if ((pidLc == -1 || pidLc == 1) && topolLcpiKp) {
+        statusLcpiKp = 1; //identified as Lc
+      }
+      */
+
+      kaonNeg = selectionPID(trackNeg1, 321);
+      prot1 = selectionPID(trackPos1, 2212);
+      prot2 = selectionPID(trackPos2, 2212);
+      pion1 = selectionPID(trackPos1, 211);
+      pion2 = selectionPID(trackPos2, 211);
+
+      if (!d_FilterPID) {
+        /// PID not applied at filtering level
+        /// set the PID flags to 1
+        pidXicPKPi = 1;
+        pidXicPiKP = 1;
+      } else {
+        /// PID applied at filtering level
+        /// Accept only candidates recognised either at least as pKpi or as piKp
+        if (kaonNeg == 1) {
+          if (std::abs(prot1) == 1 && std::abs(pion2) == 1) {
+            pidXicPKPi = 1;
+          }
+          if (std::abs(pion1) == 1 && std::abs(prot2) == 1) {
+            pidXicPiKP = 1;
+          }
+        }
+      }
+
+      if ((pidXicPKPi == 1) && topolXicpKpi) {
```

Comment:
Hi @mfaggin  . You are not including the case in which pidXicPKPi or pidXicPiKP = -1 (i.e. no PID information).
At the moment you are rejecting them and this is a very tight selection.

Reply:
Hi @DelloStritto , I think this is not true, because pidXicPKPi and pidXicPiKP are set to 1 both in case the returned value from selectionPID is +1 or -1 (look at lines 422 for example: I am checking the absolute value). Do you agree or am I wrong?

Reply:
Ah, I see. You are requiring the abs. But at this point shoudn't be the same for the kaon? (L. 421)

Reply:
Hi @DelloStritto , indeed you are right! Thanks a lot, I fixed it.